tests/test_fernet.py

```diff
@@ -25,10 +26,18 @@ def json_parametrize(keys, filename):
     vector_file = cryptography_vectors.open_vector_file(
         os.path.join("fernet", filename), "r"
     )
+
+    def _id_gen():
+        idx = itertools.count()
+        while True:
+            yield f"{filename}:{next(idx)}"
+
     with vector_file:
         data = json.load(vector_file)
         return pytest.mark.parametrize(
-            keys, [tuple([entry[k] for k in keys]) for entry in data]
+            keys,
+            [tuple([entry[k] for k in keys]) for entry in data],
+            ids=_id_gen(),
```

Comment:
What is this for?

Reply:
commit message:

> Use the vector filename and array index for the pytest id rather than a
concatenation of the vector content.
>
> eg: `tests/test_fernet.py::TestFernet::test_invalid[invalid.json:2]`

was eg:
```
tests/test_fernet.py::TestFernet::test_invalid[gAAAAAAdwJ6wAAECAwQFBgcICQoLDA0ODy021cpGVWKZ_eEwCGM4BLLF_5CV9dOPmrhuVUPgJobwOz7JcbmrR64jVmpU4IwqDA==-1985-10-26T01:20:01-07:00-60-hello-cw_0x689RpI-jtRR7oE8h_eQsKImvJapLeSbXpwF4e4=]
```

Reply:
Can you add a comment explaining that "this improves the output for readability" or something like that.